### PR TITLE
Expose job name as env var

### PR DESCRIPTION
**Pull request by edrevo**
This PR exposes the job display name as an environment variable. The job display name is exposed in webhooks as `name`([docs](https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_job)) so having it as an environment variable allows correlating data between the webhooks and processes that run in the runner.

This change would be specifically useful to improve Datadog's CI Visibility integration with GitHub Actions, where we currently require users to manually expose the job name as a variable for the integration to fully work ([docs](https://docs.datadoghq.com/continuous_integration/pipelines/custom_tags_and_measures/?tab=linux#add-tags-and-measures-to-github-jobs)).

If this change is acceptable for the runner team, I am also happy to open a PR to update the documentation around default environment variables ([docs](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)).

### CODE REVIEW

**Comment by edrevo** [comment]
Gentle ping on this one @Link- @nikola-jokic @TingluoHuang . Is there anything I can do to get this reviewed?

**Comment by edrevo** [comment]
@Link- @nikola-jokic @TingluoHuang  😢

**Comment by c3charvat** [comment]
This is a duplicate of #3489, and no one is reviewing this and @TingluoHuang closed this without merging before but hasn't replied why.

**Comment by c3charvat** [comment]
The simple fact of the matter is jobname+ matrix inputs exceeding  97 characters gets truncated... Which leads to job names that are unique on start but are not unique on the API search... 

Tldr: truncating the uuid of a job is not good. Not giving you a uuid to search for is worse.
